### ISSUE TITLE
[New]: Backup indexes

### ISSUE THREAD

**Issue by githubtlvmc** [hashicorp/terraform-provider-aws#40672]
### Description

New Data Source
Backup indexes
Backups of certain resource types, such as Amazon S3 and Amazon EBS, can be referenced and found later through a search of your backups and items.

Only backups (recovery points) with an index will be included in a search. Opt in to index creation in your backup plan to have AWS Backup automatically create an index when your recovery point is created. If the resource type is search-compatible, it can be included in your future searches.

For scheduled (periodic) backups, a new index is created or updated with each backup creation determined by the backup frequency you set in your backup plan. Continuous backups include daily index creation and updates.

+Mike Comment
My take would be to include the [IndexAction](https://docs.aws.amazon.com/aws-backup/latest/devguide/API_IndexAction.html) API as part of the rule arguments for [aws_backup_plan](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/backup_plan) and create a data source for [IndexedRecoveryPoint](https://docs.aws.amazon.com/aws-backup/latest/devguide/API_IndexedRecoveryPoint.html).

### Requested Resource(s) and/or Data Source(s)

aws_backup_Backup indexes

### Would you like to implement a fix?

None

**Comment by mike-kaminski** [comment]
My take would be to include the [IndexAction](https://docs.aws.amazon.com/aws-backup/latest/devguide/API_IndexAction.html) API as part of the rule arguments for [aws_backup_plan](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/backup_plan) and create a data source for [IndexedRecoveryPoint](https://docs.aws.amazon.com/aws-backup/latest/devguide/API_IndexedRecoveryPoint.html).

**Comment by githubtlvmc** [comment]
?

**Comment by neubauek** [comment]
Hello. I am very interested in getting S3 indexes created as part of our backups. Is there any plan on implementing this?

**Comment by KevinTrafficSupply** [comment]
?

**Comment by hasadi-ha** [comment]
Just to flag a related but distinct gap that may need to be resolved first:
before a data source for IndexedRecoveryPoint is useful, there needs to be a
way to actually create the index through Terraform in the first place.

Currently the `aws_backup_plan` resource has no `index_actions` argument on
the `rule {}` block, so there is no declarative way to enable indexing at
backup creation time. A recovery point can't be indexed via a Terraform-managed
plan until that resource-side argument exists.

I've opened a separate issue for that prerequisite change here: #48029 
Once that lands, a data source for IndexedRecoveryPoint would make full sense as a follow-on.
